app/controllers/articles_controller.rb

```diff
@@ -40,7 +40,10 @@ def destroy
   private
 
   def set_article
-    @article = Article.find(params[:id])
+    @article = Article.archived.find_by(id: params[:id])
+    if @article.nil?
+      redirect_to articles_path, alert: "You cannot edit a public or a private article"
```

Comment:
Did you tested this use case? If yes, please add a screenshot.

Reply:
<img width="693" alt="Screenshot 2021-11-30 at 1 29 01 PM" src="https://user-images.githubusercontent.com/94609278/144012016-371b30ff-0173-455c-b321-ffadeba48f0c.png">